hazelcast-jet-core/src/main/java/com/hazelcast/jet/impl/pipeline/PipelineImpl.java

```diff
@@ -158,6 +154,16 @@ public String toDotString() {
 
     void register(Transform stage, List<Transform> downstream) {
         List<Transform> prev = adjacencyMap.put(stage, downstream);
-        assert prev == null : "Double registering of a Stage with this Pipeline: " + stage;
+        assert prev == null : "Double registration of a Stage with this Pipeline: " + stage;
+    }
+
+    void makeNamesUnique() {
+        Set<String> usedNames = new HashSet<>();
+        for (Transform transform : adjacencyMap.keySet()) {
```

Comment:
is this done in topological way? so we don't end up with stages like map-3 -> map-2 -> map

Reply:
Yes, pipeline transforms are always added in topological order. You can't add sink before the source. It's also checked [here](https://github.com/hazelcast/hazelcast-jet/pull/1242/files#diff-f1c767808988b2fb6d4adb9a4971023bR107) before creating the DAG.

Reply:
Once we start modifying the transform DAG, it will disturb the assumed ordering.

Reply:
We don't do it now. That's why there's the check in `createDag`: if we ever unintentionally break the order, it will fail.